src/ucp/core/ucp_worker.c

```diff
@@ -2395,10 +2407,11 @@ static unsigned ucp_worker_discard_uct_ep_progress(void *arg)
     status = uct_ep_flush(uct_ep, req->send.discard_uct_ep.ep_flush_flags,
                           &req->send.state.uct_comp);
     if (status == UCS_ERR_NO_RESOURCE) {
+        req->send.uct.func = ucp_worker_discard_uct_ep_pending_cb;
+
         status = uct_ep_pending_add(uct_ep, &req->send.uct, 0);
         if (status == UCS_ERR_BUSY) {
-            /* try again to flush UCT EP */
-            return 0;
+            goto try_again;
```

Comment:
minor: can avoid `goto` with the same code indentation checking against UCS_OK:
```
status = uct_ep_pending_add(uct_ep, &req->send.uct, 0);
if (status == UCS_OK) {
    return 1;
}

ucs_assert(status == UCS_ERR_BUSY);
/* try again to flush UCT EP */
uct_worker_progress_register_safe(worker->uct,
                                  ucp_worker_discard_uct_ep_progress,
                                  req, UCS_CALLBACKQ_FLAG_ONESHOT,
                                  &cb_id);
return 0;
```

Reply:
removed `goto` operator, but I prefer to handle `UCS_ERR_BUSY` first and have a single return point for success/in-progress (from flush)/errors